diff -up a/texlive-20180414-source/texk/kpathsea/texmf.cnf.texinfo-fix a/texlive-20180414-source/texk/kpathsea/texmf.cnf
--- a/texlive-20180414-source/texk/kpathsea/texmf.cnf.texinfo-fix	2018-04-30 13:17:29.671610006 -0400
+++ a/texlive-20180414-source/texk/kpathsea/texmf.cnf	2018-04-30 13:18:25.434296471 -0400
@@ -82,6 +82,9 @@ TEXMFSYSCONFIG = $TEXMFROOT/texmf-config
 % to %USERPROFILE% on Windows, $HOME otherwise.
 TEXMFHOME = ~/texmf
 
+% Texinfo on Fedora lives out of $TEXMFROOT
+TEXINFOHOME = $SELFAUTODIR/share/texmf
+
 % TEXMFVAR, where texconfig/updmap/fmtutil store cached runtime data.
 TEXMFVAR = ~/.texlive2018/texmf-var
 
@@ -107,7 +110,7 @@ TEXMFAUXTREES = {}
 % The odd-looking $TEXMFAUXTREES$TEXMF... construct is so that if no auxtree is
 % ever defined (the 99% common case), no extra elements will be added to
 % the search paths. tlmgr takes care to end any value with a trailing comma.
-TEXMF = {$TEXMFAUXTREES$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,!!$TEXMFLOCAL,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFDIST}
+TEXMF = {$TEXMFAUXTREES$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,$TEXINFOHOME,!!$TEXMFLOCAL,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFDIST}
 
 % Where to look for ls-R files.  There need not be an ls-R in the
 % directories in this path, but if there is one, Kpathsea will use it.
